full-service/src/json_rpc/v2/api/wallet.rs

```diff
@@ -654,10 +655,39 @@ where
                 balance_per_token,
             }
         }
-        JsonCommandRequest::get_block { block_index } => {
-            let (block, block_contents) = service
-                .get_block_object(block_index.parse::<u64>().map_err(format_error)?)
-                .map_err(format_error)?;
+        JsonCommandRequest::get_block {
+            block_index,
+            txo_public_key,
+        } => {
+            let (block, block_contents) = match (block_index, txo_public_key) {
+                (None, None) => {
+                    return Err(format_invalid_params_error(
+                        "Must specify either block_index or txo_public_key",
+                    ));
+                }
+                (Some(_), Some(_)) => {
+                    return Err(format_invalid_params_error(
+                        "Must specify either block_index or txo_public_key, not both",
+                    ))
+                }
+                (None, Some(txo_public_key)) => {
+                    let public_key_bytes = hex::decode(txo_public_key).map_err(format_error)?;
+                    let public_key: CompressedRistrettoPublic = public_key_bytes
+                        .as_slice()
+                        .try_into()
+                        .map_err(format_error)?;
+                    let block_index = service
+                        .get_block_index_from_txo_public_key(&public_key)
+                        .map_err(format_error)?;
+                    service
+                        .get_block_object(block_index)
+                        .map_err(format_error)?
+                }
+                (Some(block_index), None) => service
+                    .get_block_object(block_index.parse::<u64>().map_err(format_error)?)
+                    .map_err(format_error)?,
```

Comment:
This will return ```LedgerDB(Record not found)``` as the error response. format_error is just formatting the specific error as JsonRPCError::server_error

Reply:
This is the full response from this particular error

```
{
    "method": "get_block",
    "error": {
        "code": -32603,
        "message": "InternalError",
        "data": {
            "server_error": "LedgerDB(Record not found)",
            "details": "Error with LedgerDB: Record not found"
        }
    },
    "jsonrpc": "2.0",
    "id": 1
}
```

Reply:
Ah I see, cool.